src/Fhir.Anonymizer.Tool/FhirResourceDataProcessor.cs

```diff
@@ -74,33 +77,34 @@ public void AnonymizeBulkDataFolder(string inputFolder, string outputFolder, boo
                     Directory.CreateDirectory(resourceOutputFolder);
                 }
 
-                var processedCount = 0;
-                var processedErrorCount = 0;
+                int completedCount = 0;
+                int failedCount = 0;
+                int consumeCompletedCount = 0;
                 using (FileStream inputStream = new FileStream(bulkResourceFileName, FileMode.Open))
                 using (FileStream outputStream = new FileStream(bulkResourceOutputFileName, FileMode.Create))
                 {
-                    using StreamReader reader = new StreamReader(inputStream);
-                    using StreamWriter writer = new StreamWriter(outputStream);
+                    using FhirStreamReader reader = new FhirStreamReader(inputStream);
+                    using FhirStreamConsumer consumer = new FhirStreamConsumer(outputStream);
+                    Func<string, string> anonymizeFunction = (content) => _engine.AnonymizeJson(content);
+
+                    Stopwatch stopWatch = new Stopwatch();
+                    stopWatch.Start();
 
-                    string resourceLine;
-                    string resultLine;
-                    while ((resourceLine = reader.ReadLine()) != null)
+                    FhirPartitionedExecutor executor = new FhirPartitionedExecutor(reader, consumer, anonymizeFunction);
+                    executor.PartitionCount = Environment.ProcessorCount;
```

Comment:
Should that be ProcessorCount*2?

Reply:
Tested *2 improve performance a bit for core count > 4. Updated to * 2. Thanks